### PR TITLE
balance: add some metrics

### PR DESCRIPTION
**Pull request by nolouch**
PTAL @disksing @siddontang

### CODE REVIEW

File: server/metrics.go [hunk 1]
```diff
@@ -56,6 +56,14 @@ var (
...
 			Help:      "Counter of system time jumps backward.",
 		})
+
+	schedulerStatusGauge = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "pd",
+			Subsystem: "scheduler",
+			Name:      "Status",
+			Help:      "Status of the scheduler.",
+		}, []string{"group", "type"})
```
Comment (disksing) [inline server/metrics.go]: Maybe use 2 different metrics? 1 for scheduler limits, 1 for hotspot.

File: server/coordinator.go [hunk 2]
```diff
@@ -158,6 +159,39 @@ func (c *coordinator) getSchedulers() []string {
...
+	c.RLock()
+	defer c.RUnlock()
+	for _, s := range c.schedulers {
+		var allowScheduler float64
+		if s.AllowSchedule() {
+			allowScheduler = 1
+		}
+		limit := float64(s.GetResourceLimit())
+
+		schedulerStatusGauge.WithLabelValues(s.GetName(), "allow_schedule").Set(allowScheduler)
```
Comment (siddontang) [inline server/coordinator.go]: s/allow_schedule/allowed/

File: server/metrics.go [hunk 3]
```diff
@@ -56,6 +56,22 @@ var (
...
+			Help:      "Status of the scheduler.",
+		}, []string{"kind", "type"})
+
+	hotSpotStatusGauge = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: "pd",
+			Subsystem: "hotspot",
+			Name:      "status",
+			Help:      "Status of the hotspot.",
+		}, []string{"group", "type"})
```
Comment (siddontang) [inline server/metrics.go]: store may be better than group?

File: server/coordinator.go [hunk 4]
```diff
@@ -158,6 +159,39 @@ func (c *coordinator) getSchedulers() []string {
...
+	if !ok {
+		return
+	}
+	status := s.Scheduler.(*balanceHotRegionScheduler).GetStatus()
+	for storeID, stat := range status {
+		group := fmt.Sprintf("store_%d", storeID)
+		totalWriteBytes := float64(stat.TotalWrittenBytes)
+		hotWriteRegionCount := float64(stat.RegionCount)
+
+		hotSpotStatusGauge.WithLabelValues(group, "total_write_bytes").Set(totalWriteBytes)
```
Comment (siddontang) [inline server/coordinator.go]: total_written_bytes

**Comment by disksing** [comment]
LGTM.

**Pull request merged**